### PR TITLE
Correct colours for all renderers

### LINKED ISSUE #550: Orientation information in the renderer
Once the orientation labelling (#243) is done, we need to display this information in the renderered diagram.

Here are some ideas:
- different colours (how will this interact with the different shades for points, wires, regions)
- asymmetric boxes for the points (related to #519)
- arrows on the wires (#553)
- dashed/dotted wires?
- flip normals in GL

### PR DESCRIPTION
**Pull request by thud**
Closes #550 once and for all, implementing correct colours for 3D, 4D and remaining 2D renderers.

Builds on #717.

### CODE REVIEW

**Comment by thud** [comment]
I have done my best to implement correct colours for the GL renderer. However, I'm not 100% sure that the renderer works as expected. I would be grateful for some testing if anyone wants to try it.

File: homotopy-graphics/src/geom.rs [hunk 1]
```diff
@@ -98,6 +98,7 @@ pub struct VertData {
     pub position: Vec4,
     pub boundary: [bool; 4],
     pub generator: Generator,
+    pub k: usize,
```
Comment (doctorn) [inline homotopy-graphics/src/geom.rs:101]: Perhaps you could introduce a typealias or new type for this `usize`? What's the maximum value likely to be (could we get away with a `u32` or `u16`)? `k` is also not a very informative name haha

**Review by doctorn — CHANGES_REQUESTED** [review]
Well done navigating the buffering code! Could you move `buffers.rs` into `src/diagram_gl`?

File: homotopy-graphics/src/geom.rs (hunk 1 above)
Comment (thud) [reply homotopy-graphics/src/geom.rs:101]: Yeah very true. I'll get something better implemented soon.

File: homotopy-graphics/src/style.rs [hunk 2]
```diff
@@ -51,16 +46,16 @@ impl Color {
     // Get color in lightened form. We wrap colors if needed so that they are sufficiently
     // distinguishable for end users.
     //
-    // Colors are calculated from C and R where:
+    // Colors are distinguished as pairs (C, R) where:
     //      C = max(0, D - N - K)
```
Comment (doctorn) [inline homotopy-graphics/src/style.rs:50]: Could you `println` the values of $C$ for some diagram? $K = max(0, D - N)$ so I have a feeling this is always $0$.

Comment (thud) [reply homotopy-graphics/src/style.rs:50]: In think $K$ is technically not defined a $\max(0, D - N)$. It's actually inferred from the representation of a generator which comes out of the core graphics code.

Comment (thud) [reply homotopy-graphics/src/style.rs:50]: 2022-09-05-082305_786x419_scrot

In this case, `Cell 2` gives $C = 0$ but `Cell 0` gives $C = 1$.

`hom` attached: [c_values.zip](https://github.com/homotopy-io/homotopy-rs/files/9487626/c_values.zip)

**Pull request merged**